sklearn/preprocessing/data.py

```diff
@@ -2579,6 +2579,181 @@ def quantile_transform(X, axis=0, n_quantiles=1000,
                          " axis={}".format(axis))
 
 
+class PowerTransformer(BaseEstimator, TransformerMixin):
+    """Apply a power transform featurewise to make data more Gaussian-like.
+
+    Power transforms are a family of parametric, monotonic transformations
+    that are applied to make data more Gaussian-like. This is useful for
+    modeling issues related to heteroscedasticity (non-constant variance),
+    or other situations where normality is desired. Note that power
+    transforms do not result in standard normal distributions.
+
+    Currently, PowerTransformer supports the Box-Cox transform. Box-Cox
+    requires input data to be strictly positive. The optimal parameter
+    for minimizing skewness is estimated through maximum likelihood.
+
+    Read more in the :ref:`User Guide <preprocessing_transformer>`.
+
+    Parameters
+    ----------
+    method : str, (default='boxcox')
```

Comment:
So without the hyphen, we'll support yeojohnson in the future?

Reply:
I think that would be fine! I can't decide on one myself. On one hand, `box-cox` and `yeo-johnson` look cleaner, but on the other hand, I'm not sure about hyphens in parameters...

Reply:
Okay. I've changed my mind. Let's go for cleaner.